mempool/src/shared_mempool.rs

```diff
@@ -427,6 +542,15 @@ async fn inbound_network_task<V>(
     let workers_available = smp.config.shared_mempool_max_concurrent_inbound_syncs;
     let bounded_executor = BoundedExecutor::new(workers_available, executor);
 
+    let mut consensus_events = match consensus_events {
```

Comment:
really don't like it
we can probably remove Option part. Cause in follow up diff you'll have to send events from state sync to mempool and it's always available on both full nodes and validators